pkg/klusterlet/clusterca/clusterca_controller.go

```diff
@@ -0,0 +1,126 @@
+package clusterca
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	clusterv1beta1 "github.com/open-cluster-management/multicloud-operators-foundation/pkg/apis/internal.open-cluster-management.io/v1beta1"
+	"github.com/open-cluster-management/multicloud-operators-foundation/pkg/utils"
+	openshiftclientset "github.com/openshift/client-go/config/clientset/versioned"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	UpdateInterval = 60 * time.Second
+)
+
+type ClusterCaReconciler struct {
+	client.Client
+	OpenshiftClient openshiftclientset.Interface
+	KubeClient      kubernetes.Interface
+	ClusterName     string
+}
+
+func (r *ClusterCaReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	managedClusterInfo := &clusterv1beta1.ManagedClusterInfo{}
+	err := r.Client.Get(ctx, req.NamespacedName, managedClusterInfo)
+	if err != nil {
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
+	newStatus := managedClusterInfo.DeepCopy().Status
+	if newStatus.DistributionInfo.Type != clusterv1beta1.DistributionTypeOCP {
+		return ctrl.Result{}, nil
+	}
+
+	//get ocp apiserver url
+	kubeAPIServer, err := utils.GetKubeAPIServerAddress(r.OpenshiftClient)
+	if err != nil {
+		klog.Errorf("Failed to get kube Apiserver. err:%v", err)
+		return ctrl.Result{}, err
+	}
+
+	//get ocp ca
+	clusterca := r.getClusterCA(kubeAPIServer)
+	if len(clusterca) <= 0 {
+		klog.Errorf("Failed to get clusterca.")
+		return ctrl.Result{}, fmt.Errorf("failed to get clusterca")
+	}
+
+	klog.V(5).Infof("kubeapiserver: %v, clusterca:%v", kubeAPIServer, clusterca)
+
+	newConfig := clusterv1beta1.ClientConfig{
+		URL:      kubeAPIServer,
+		CABundle: clusterca,
+	}
+
+	if reflect.DeepEqual(newStatus.DistributionInfo.OCP.ManagedClusterClientConfig, newConfig) {
+		return ctrl.Result{RequeueAfter: UpdateInterval}, nil
```

Comment:
why requeu here?

Reply:
The reconcile need re queue  periodically, because it need to check if the ca changed periodically, and then update it.